timemachine/fe/free_energy.py

```diff
@@ -392,12 +428,65 @@ def _combine(self, ligand_values, host_values=None):
         return np.concatenate([host_values, ligand_values])
 
 
-def get_context(initial_state: InitialState) -> Context:
-    bound_impls = [p.to_gpu(np.float32).bound_impl for p in initial_state.potentials]
+def get_water_sampler_params(initial_state: InitialState) -> NDArray:
+    """Given an initial state, return the parameters that define the parameters of water-environment parameters
+
+    Since we split the different components of the NB into ligand-water, ligand-protein, we want to use the ligand-water parameter
+    to ensure that the ligand component is correctly configured for the specific lambda window.
+    """
+    summed_pot = next(p.potential for p in initial_state.potentials if isinstance(p.potential, SummedPotential))
+    # TBD Figure out a better way of handling the jankyness of having to select the IxnGroup that defines the Ligand-Water
+    # Currently this just hardcodes to the first ixn group potential which is the ligand-water ixn group as returned by HostTopology.parameterize_nonbonded
```

Comment:
@jkausrelay - triple check this?

Reply:
This is correct for the current implementation (https://github.com/proteneer/timemachine/blob/05b1a7a787ed4bae0087926fdaf820f4ada648dc/timemachine/fe/topology.py#L776). @badisa  Can you add a note there to make sure to update the water sampling code if this is changed?

Reply:
Added comment to the topology file in https://github.com/proteneer/timemachine/pull/1286/commits/aeadaa0be1fd1430d9e5d816fb41d2d08831ee26